content/rooms/fragments/v2-state-res.md

```diff
@@ -82,27 +82,41 @@ selecting, among all the candidate vertices, the smallest vertex using
 the above comparison relation.
 
 **Mainline ordering.**
-Given an `m.room.power_levels` event *P*, the *mainline of* *P* is the
-list of events generated by starting with *P* and recursively taking the
-`m.room.power_levels` events from the `auth_events`, ordered such that
-*P* is last. Given another event *e*, the *closest mainline event to*
-*e* is the first event encountered in the mainline when iteratively
-descending through the `m.room.power_levels` events in the `auth_events`
-starting at *e*. If no mainline event is encountered when iteratively
-descending through the `m.room.power_levels` events, then the closest
-mainline event to *e* can be considered to be a dummy event that is
-before any other event in the mainline of *P* for the purposes of
-condition 1 below.
+Let *P* = *P*<sub>0</sub> be an `m.room.power_levels` event.
+Starting with *i* = 0, repeatedly fetch *P*<sub>*i*+1</sub>, the
+`m.room.power_levels` event in the `auth_events` of *P<sub>i</sub>*.
+Increment *i* and repeat until *P<sub>i</sub>* has no `m.room.power_levels`
+event in its `auth_events`.
+The *mainline of P*<sub>0</sub> is the list of events
+    [*P*<sub>n</sub> , ... , *P*<sub>1</sub>, *P*<sub>0</sub>],
+ordered from oldest to newest.
```

Comment:
In the light of https://github.com/matrix-org/matrix-spec/pull/1107/files#r892668039, how about this?

```suggestion
The *mainline of P*<sub>0</sub> is the list of events
    [*P*<sub>0</sub> , *P*<sub>1</sub>, ... , *P<sub>n</sub>*],
fetched in this way.
```

Reply:
I don't follow. This seems to reverse the order of the list?

Reply:
though actually we don't care about the order of the list. so yes, lgtm.

Reply:
> though actually we don't care about the order of the list. so yes, lgtm.

Yeah, that's one of the benefits of having explicit indices. I think it's helpful that the two lists in this section go in the same direction. To make that change, we pay the cost of flipping the ordering condition (1) below. (Before: prefer events whose PL ancestor appears _earlier_ in the mainline. After: prefer events with _greater_ mainline position.)